src/styles/functions/_set-color.scss

```diff
@@ -0,0 +1,17 @@
+@function set-color($color) {
+  @if (lightness($color) > 50) {
+    @return $color-ui-black;
+  } @else {
+    @return $color-ui-white;
+  }
+}
+
+@mixin color-variants {
```

Comment:
@Jonic Is this the sort of thing you meant?

Could move this to a mixins directory?

Reply:
Yeah, I think we should keep mixins and functions separate from one another.

As for what THIS function does - we can't anticipate that asking for color variants will always apply to `backgroud-color`, `border-color`, and a tweaked `color`. My take on this was that we'd pass a property in, like so: `color-variants('background-color');`.

We could maybe have a flag as well that optionally triggers the setting of `color` if we pass `true` as a second arg, or maybe even if it detects we're setting the `background-color`.

It's really close to how I imagined it, but I think right now the only place it would be useful is for the buttons, so I think it could do with being a little more reusable :)

I wonder if perhaps we should think about writing tests for these things too? Looks like it's doable! https://www.sitepoint.com/testing-sass-function-5-minutes/

Reply:
@Jonic Tests look good.

I think what you're suggesting adds a lot of complexity for very little value at the moment, and possibly ever, so I'd like to revert to my first version of this:

```
@each $color-name, $color-value in $colors {
  .#{'c-button--' + $color-name} {
    background-color: #{$color-value};
    border-color:          #{$color-value};
    color:                      set-color($color-value);
  }
}
```

I generally feel it's much better to be straightforward & explicit rather than adding complication and optimisation too early. https://martinfowler.com/bliki/Yagni.html

Reply:
Yeah that's fair. I think this is less a case of "YAGNI" and more a case of "YDNIY". I think we'll need something like that as the library grows, but right now it's premature optimisation, yeah. I've used it a LOT on the projects site (where everything has color variants), but right now there's no real need for it to be in Bits. If you make that revert then I'll review again and approve :)